File: Cpp implementation/Code.cpp
#include <iostream>
#include <fstream>
using namespace std;
void swap(int &a, int &b) {
    int temp = a;
    a = b;
    b = temp;
}
void combSort(int arr[], int n) {
    int gap = n;
    float shrink = 1.3;
    bool swapped = true;

    while (gap > 1 || swapped) {
        gap = (int)(gap / shrink);
        if (gap < 1)
            gap = 1;

        swapped = false;

        for (int i = 0; i < n - gap; i++) {
            if (arr[i] > arr[i + gap]) {
                swap(arr[i], arr[i + gap]);
                swapped = true;
            }
        }
    }
}
void printArrayToFile(int arr[], int size, const string& filename) {
    ofstream outFile(filename, ios::app);

    if (!outFile) {
        cerr << "Error: Unable to open the file " << filename << endl;
        return;
    }

    for (int i = 0; i < size; i++)
    {
        outFile << arr[i] << " ";
    }
    outFile << endl;
    outFile.close();

}
void readTestCases(const string& filename) {
    ifstream inFile(filename);

    if (!inFile) {
        cerr << "Error: Unable to open the file " << filename << endl;
        return;
    }

    int n;

    while (inFile >> n) {
        int *arr = new int[n];
        for (int i = 0; i < n; i++)
            inFile >> arr[i];
        combSort(arr, n);
        printArrayToFile(arr, n, "output.txt");
        delete[] arr;
    }
    cout<<"Sorted array has been written to output.txt"<<endl;

    inFile.close();
}

int main() {
    readTestCases("input.txt");

    return 0;
}

File: Python implementation/Code.py
def swap(arr, i, j):
    arr[i], arr[j] = arr[j], arr[i]
def combSort(arr):
    gap = len(arr)
    shrink = 1.3
    swapped = True

    while gap > 1 or swapped:
        gap = int(gap / shrink)
        if gap < 1:
            gap = 1

        swapped = False

        for i in range(len(arr) - gap):
            if arr[i] > arr[i + gap]:
                swap(arr, i, i + gap)
                swapped = True
def printArrayToFile(arr, filename):
    with open(filename, 'w') as outFile:
        for array in arr:
            outFile.write(' '.join(map(str, array)) + '\n')
    print("Sorted arrays have been written to", filename)

if __name__ == "__main__":
    try:
        with open("input.txt","r") as file:
            arrays = []
            lines = file.readlines()
            for line in lines:
                arr = list(map(int, line.split()))[1:] 
                arrays.append(arr)
    except FileNotFoundError:
        print("Error: Unable to open the file input.txt")
        exit(1)
    for arr in arrays:
        combSort(arr)
    printArrayToFile(arrays, "output.txt")
